### PR TITLE
CLI: if user misprinted with the cli-argument - save should make a normal logging instead of 'directory not found'

### LINKED ISSUE #253: CLI: if user misprinted with the cli-argument - save should make a normal logging instead of 'directory not found'
If user has made a mistake and misprinted:

```
$ ./save.kexe --repoTR-type plain save
[ERROR]: Not able to find directory '--repoTR-type'. Please provide a valid path to the root directory of test files.
```

Save should help a user and print help. Actually cli library that is used in save is doing that, but the usage is incorrect.

### PR DESCRIPTION
**Pull request by Cheshiriks**
What's done:
* Fixed IndexOutOfBoundsException
Closes #253

### CODE REVIEW

File: save-core/src/commonNonJsMain/kotlin/org/cqfn/save/core/Save.kt [hunk 1]
```diff
@@ -55,46 +55,51 @@ class Save(
...
+                    // execute created plugins
+                    executePlugin(it, reporter)
+                }
+                ?.also {
+                    reporter.onSuiteEnd(testConfig.getGeneralConfig()?.suiteName!!)
+                }
+        }
+        val saveToml = testConfigs.map { it.getGeneralConfig()?.configLocation }
+        val excluded = testConfigs.map { it.getGeneralConfig()?.excludedTests?.toMutableList() ?: emptyList() }
+            .reduceOrNull { acc, list -> acc + list }
```
Comment (akuleshov7) [inline save-core/src/commonNonJsMain/kotlin/org/cqfn/save/core/Save.kt:97]: val excludedNote = excluded?.let { 
    Note: please check excludedTests: $excluded in following save.toml files: $saveToml
} ?: ""

File: save-cli/src/nativeMain/kotlin/org/cqfn/save/cli/SaveConfigReader.kt [hunk 2]
```diff
@@ -26,9 +26,13 @@ private val fs: FileSystem = FileSystem.SYSTEM
  * @return this config in case we have valid configuration
  */
 fun SaveProperties.validate(): SaveProperties {
-    this.testFiles ?: logErrorAndExit(ExitCodes.INVALID_CONFIGURATION,
-        "`test files list in CLI is missing or null. " +
-                "Save is not able to start processing without an information about the tests that should be run.")
+    if (this.testFiles.isNullOrEmpty()) {
+        logErrorAndExit(
+            ExitCodes.INVALID_CONFIGURATION,
+            "test files list in CLI is missing or null. " +
```
Comment (akuleshov7) [inline save-cli/src/nativeMain/kotlin/org/cqfn/save/cli/SaveConfigReader.kt:32]: Test files

File: save-cli/src/nativeMain/kotlin/org/cqfn/save/cli/SaveConfigReader.kt [hunk 3]
```diff
@@ -26,9 +26,13 @@ private val fs: FileSystem = FileSystem.SYSTEM
...
+                    "Save is not able to start processing without an information about the tests or test root path that should be run."
```
Comment (akuleshov7) [inline save-cli/src/nativeMain/kotlin/org/cqfn/save/cli/SaveConfigReader.kt:33]: that should be used for exesution

File: save-cli/src/nativeMain/kotlin/org/cqfn/save/cli/SaveConfigReader.kt [hunk 4]
```diff
@@ -139,6 +143,8 @@ private fun errorAndExitNotFoundDir() {
 
 private fun errorAndExitNotValidDir(testRootPath: Path) {
     logErrorAndExit(
-        ExitCodes.INVALID_CONFIGURATION, "Not able to find directory '$testRootPath'." +
-                " Please provide a valid path to the root directory of test files.")
+        ExitCodes.INVALID_CONFIGURATION,
+        "Save parsed the argument '$testRootPath' that you have provided to cli as a root for test directory and is not able to find it. " +
+                "Please provide a valid path to the root directory of test files. " +
+                "If you wanted to pass a configuration option instead, please check the list of available options using '—help'.")
```
Comment (akuleshov7) [inline save-cli/src/nativeMain/kotlin/org/cqfn/save/cli/SaveConfigReader.kt:149]: --help

**Review by akuleshov7 — CHANGES_REQUESTED** [review]
fix small notes

File: save-core/src/commonNonJsMain/kotlin/org/cqfn/save/core/Save.kt (hunk 1 above)
Comment (Cheshiriks) [reply save-core/src/commonNonJsMain/kotlin/org/cqfn/save/core/Save.kt:97]: Fixed

**Review by akuleshov7 — APPROVED** [review]

**Pull request merged**